<a href="https://colab.research.google.com/github/singh2jz/Project_2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

# First Choice

In [2]:
heart_df = pd.read_csv('/content/drive/MyDrive/Coding Dojo/Week 7/Data/heart.csv')

In [3]:
heart_df.head()

Age Sex ChestPainType  ...  Oldpeak  ST_Slope  HeartDisease
0   40   M           ATA  ...      0.0        Up             0
1   49   F           NAP  ...      1.0      Flat             1
2   37   M           ATA  ...      0.0        Up             0
3   48   F           ASY  ...      1.5      Flat             1
4   54   M           NAP  ...      0.0        Up             0

[5 rows x 12 columns]

In [5]:
heart_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB




1.   https://www.kaggle.com/fedesoriano/heart-failure-prediction
2.   Dataset is composed of features that can help predict heart disease. The goal is to have some type of early warning system for people that are high risk.
3.   Target is 'HeartDisease'
4.   Classification
5.   11 features
6.   918 rows
7.   There aren't any nulls but still need to check the data for outliers. Will need to check if data is balanced, might need to make some changes if severely unbalanced.



## Second Choice

In [6]:
stroke_df = pd.read_csv('/content/drive/MyDrive/Coding Dojo/Week 7/Data/healthcare-dataset-stroke-data.csv')

In [7]:
stroke_df.head()

id  gender   age  ...   bmi   smoking_status stroke
0   9046    Male  67.0  ...  36.6  formerly smoked      1
1  51676  Female  61.0  ...   NaN     never smoked      1
2  31112    Male  80.0  ...  32.5     never smoked      1
3  60182  Female  49.0  ...  34.4           smokes      1
4   1665  Female  79.0  ...  24.0     never smoked      1

[5 rows x 12 columns]

In [8]:
stroke_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


1.   https://www.kaggle.com/fedesoriano/stroke-prediction-dataset
2.   I chose this dataset as my second choice because it has a similar goal to my first choice. This data is used to predict if a patient is likely to have a stroke. 
3.   Target is 'stroke'
4.   Classification
5.   10 features
6.   5110 rows
7.   I see bmi has nulls. Will need to check for balance not only in target but some other features. Also need to check id for duplicates(maybe a patient tested at a different time in life).